File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/eksportering/NæringOgNæringskode5siffer.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.eksportering;

public class NæringOgNæringskode5siffer {
    String næring;
    String næringskode5Siffer;

    public NæringOgNæringskode5siffer(
            String næring,
            String næringskode5Siffer
    ) {
        this.næring = næring;
        this.næringskode5Siffer = næringskode5Siffer;
    }


    public String getNæring() { return næring; }
    public String getNæringskode5Siffer() { return næringskode5Siffer; }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        NæringOgNæringskode5siffer that = (NæringOgNæringskode5siffer) o;

        if (!næring.equals(that.næring)) return false;
        return næringskode5Siffer.equals(that.næringskode5Siffer);
    }

    @Override
    public int hashCode() {
        int result = næring.hashCode();
        result = 31 * result + næringskode5Siffer.hashCode();
        return result;
    }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/NæringOgNæringskode5siffer.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

class NæringOgNæringskode5siffer(@JvmField var næring: String, @JvmField var næringskode5Siffer: String) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as NæringOgNæringskode5siffer
        return if (næring != that.næring) false else næringskode5Siffer == that.næringskode5Siffer
    }

    override fun hashCode(): Int {
        var result = næring.hashCode()
        result = 31 * result + næringskode5Siffer.hashCode()
        return result
    }
}
